### PR TITLE
[14.0][MIG] connector

### PR DESCRIPTION
**Pull request by eantones**
Depends on:
- [x] [OCA/queue#263](https://github.com/OCA/queue/pull/263)
- [x] [OCA/connector#382](https://github.com/OCA/connector/pull/382)
- [x] [OCA/connector#384](https://github.com/OCA/connector/pull/384)

### CODE REVIEW

**Comment by eantones** [comment]
All tests passed perfectly on my test environment 
I think there's something wrong with the test-dependencies.txt and travis is not able to clone the queue_job PR. 
Any ideas?
Thanks! :-)

File: test-requirements.txt
Comment (simahawk) [inline test-requirements.txt]: addon name seems wrong: `s/_/-` > odoo14-addon-queue-job

Comment (eantones) [reply test-requirements.txt]: @simahawk how do I know the name if it's not from the module folder name https://github.com/OCA/queue/tree/13.0/queue_job? Why with dash instead of underscore?

Comment (eantones) [reply test-requirements.txt]: Just tried it, same error

image

Comment (simahawk) [reply test-requirements.txt]: the URL is wrong as well, compare it w/ the others right above ;)

BTW when you depend on other PRs please reference them in the description using a list of checkboxes to flag the ones that got merged. Example:

Depends on:

- [ ] https://github.com/OCA/queue/pull/263
...

Comment (simahawk) [reply test-requirements.txt]: > @simahawk how do I know the name if it's not from the module folder name https://github.com/OCA/queue/tree/13.0/queue_job? Why with dash instead of underscore?

the name is exactly the same but with `-` instead of `_`. The format is `odoo{odoo_version}-addon-{normalized-addon-name}`.

Examples https://wheelhouse.odoo-community.org/oca-simple/ or https://pypi.org/search/?q=odoo13-addon

Comment (eantones) [reply test-requirements.txt]: > the URL is wrong as well, compare it w/ the others right above ;)

Silly mistake! Thanks to make me aware of it ;-)

Comment (eantones) [reply test-requirements.txt]: > BTW when you depend on other PRs please reference them in the description using a list of checkboxes to flag the ones that got merged. Example:
> 
> Depends on:
> 
> * [ ]  [OCA/queue#263](https://github.com/OCA/queue/pull/263)
>   ...

Done!

Comment (eantones) [reply test-requirements.txt]: > > @simahawk how do I know the name if it's not from the module folder name https://github.com/OCA/queue/tree/13.0/queue_job? Why with dash instead of underscore?
> 
> the name is exactly the same but with `-` instead of `_`. The format is `odoo{odoo_version}-addon-{normalized-addon-name}`.
> 
> Examples https://wheelhouse.odoo-community.org/oca-simple/ or https://pypi.org/search/?q=odoo13-addon

The name of the module was already correct it must be with underscore odoo14-addon-queue_job @ git+https://github.com/OCA/queue/pull/263/head#subdirectory=setup/queue_job
My mistake was just forgetting the "@refs" on the url. 
It's green now :-)
Thank you very much

**Comment by sebastienbeau** [comment]
@eantones Dependency have been merged
Can you update you commit to remove the test-dependency file ?
Thanks

**Comment by eantones** [comment]
@sebastienbeau done!
Thanks!

File: connector/models/checkpoint.py
Comment (guewen) [inline connector/models/checkpoint.py]: Maybe the good time to remove this? https://github.com/OCA/connector/issues/351

Comment (sebastienbeau) [reply connector/models/checkpoint.py]: good idea !

**Comment by esakarya** [comment]
any update ?

**Comment by kittiu** [comment]
@eantones please check travis and rebase. All dependencies are merged.

**Comment by eantones** [comment]
@kittiu, it's done. Travis is green now.

File: connector/models/checkpoint.py
Comment (kittiu) [reply connector/models/checkpoint.py]: Can this be removed as suggested?

File: connector/AUTHORS
Comment (simahawk) [inline connector/AUTHORS]: Is there a reason to keep this? If not, could you get rid of this and use `readme` fragments? The only author here is @guewen the rest goes to contributors.

File: connector/__manifest__.py
Comment (simahawk) [inline connector/__manifest__.py]: "Openerp Connector Core Editors" AFAIK there's no such thing

File: connector/__manifest__.py
Comment (simahawk) [inline connector/__manifest__.py]: "-2017" can be removed everywhere

**Review by simahawk — COMMENTED** [review]
LG, just minor things to cleanup.
Thanks!

File: connector/AUTHORS
Comment (eantones) [reply connector/AUTHORS]: I think you're right. I changed that.

File: connector/__manifest__.py
Comment (eantones) [reply connector/__manifest__.py]: done!

File: connector/__manifest__.py
Comment (eantones) [reply connector/__manifest__.py]: Done!

**Comment by eantones** [comment]
@simahawk please, check if there's more changes to do. Thanks :-)

File: connector/models/checkpoint.py
Comment (simahawk) [reply connector/models/checkpoint.py]: I think so, nobody is using it AFAIK.
@eantones likely this is the last bit :wink:

**Review by rvalyi — APPROVED** [review]
I checked that the dependencies didn't change much between v13 and v14. Also inside the module components and models have no reason to change to accomodate to any change in Odoo. So given tests are passing, that's LGTM.

**Review by rousseldenis — APPROVED** [review]
Code review. Maybe increasing a little bit coverage should be great.

@OCA/connector-maintainers We certainly should switch to main branches including non migrated modules from past versions in order to avoid such PR with 721 commits.

**Comment by guewen** [comment]
> @OCA/connector-maintainers We certainly should switch to main branches including non migrated modules from past versions in order to avoid such PR with 721 commits.

+1 
I did this for the previous versions, but I missed the occasion to ensure this for 14.0

**Review by hparfr — APPROVED** [review]

**Comment by guewen** [comment]
@eantones I'm going to propose a PR on your PR to remove checkpoints

**Comment by guewen** [comment]
@eantones can you include https://github.com/nuobit/connector/pull/1 please? Then we are good to merge :)

**Comment by eantones** [comment]
@guewen done! :-)

**Comment by eantones** [comment]
@guewen @rousseldenis regarding the 721 commits, you're right, there's too many, I can make a squash

**Comment by rousseldenis** [comment]
>  I can make a squash

No! You'll loose the commit history and that is not wanted.

**Comment by rousseldenis** [comment]
> @guewen @rousseldenis regarding the 721 commits, you're right, there's too many, I can make a squash

There are two ways of doing main branches :

- Creating the branch with no module at all from scratch (actual OCA convention apart some repos). All migration PR are created with the whole commit history like here.
- Creating the new main branch (14.0) based on previous one (13.0). So, the commit history is preserved between them. The advantage is that migration PR would contain only the new commits and is more readable.

**Comment by eantones** [comment]
@rousseldenis yes I know, so how do you suggest to reduce the commits?

**Comment by rousseldenis** [comment]
> @rousseldenis yes I know, so how do you suggest to reduce the commits?

Nope, this will be merged as is, it's ok

**Comment by eantones** [comment]
@rousseldenis ok, so you meant that we cannot do anything now because this should have been done on 14.0 branch creation right?

**Review by guewen — APPROVED** [review]
Thanks :heart: 
Waiting for the build before merging.

**Comment by guewen** [comment]
> ok, so you meant that we cannot do anything now because this should have been done on 14.0 branch creation right?

yep

**Comment by eantones** [comment]
Thanks :-)

**Comment by simahawk** [comment]
> > @guewen @rousseldenis regarding the 721 commits, you're right, there's too many, I can make a squash
> 
> There are two ways of doing main branches :
> 
>     * Creating the branch with no module at all from scratch (actual OCA convention apart some repos). All migration PR are created with the whole commit history like here.
> 
>     * Creating the new main branch (14.0) based on previous one (13.0). So, the commit history is preserved between them. The advantage is that migration PR would contain only the new commits and is more readable.

Workaround: before submitting a migration PR open another one where the module is ported as is from another version and made uninstallable.

**Comment by guewen** [comment]
Hi, I realize that test_component and test_connector add-ons are not included. You may have missed them, but they are what ensure we have no regression

**Pull request merged**
